pkg/cmd/pr/create/create.go

```diff
@@ -330,71 +352,101 @@ func createRun(opts *CreateOptions) (err error) {
 		}
 	}
 
-	if !opts.TitleProvided {
-		err = shared.TitleSurvey(opts.Prompter, state)
-		if err != nil {
-			return
-		}
+	action := shared.SubmitAction
+	if opts.IsDraft {
+		action = shared.SubmitDraftAction
 	}
 
-	defer shared.PreserveInput(opts.IO, state, &err)()
+	tpl := shared.NewTemplateManager(client.HTTP(), ctx.BaseRepo, opts.Prompter, opts.RootDirOverride, opts.RepoOverride == "", true)
 
-	if !opts.BodyProvided {
-		templateContent := ""
-		if opts.RecoverFile == "" {
-			tpl := shared.NewTemplateManager(client.HTTP(), ctx.BaseRepo, opts.Prompter, opts.RootDirOverride, opts.RepoOverride == "", true)
+	if opts.EditorMode {
+		if opts.Template != "" {
 			var template shared.Template
-
-			if opts.Template != "" {
-				template, err = tpl.Select(opts.Template)
-				if err != nil {
-					return
-				}
-			} else {
-				template, err = tpl.Choose()
-				if err != nil {
-					return
-				}
+			template, err = tpl.Select(opts.Template)
+			if err != nil {
+				return
 			}
-
-			if template != nil {
-				templateContent = string(template.Body())
+			if state.Title == "" {
+				state.Title = template.Title()
 			}
+			state.Body = string(template.Body())
 		}
 
-		err = shared.BodySurvey(opts.Prompter, state, templateContent)
+		state.Title, state.Body, err = opts.TitledEditSurvey(state.Title, state.Body)
 		if err != nil {
 			return
 		}
-	}
+		if state.Title == "" {
+			err = fmt.Errorf("title can't be blank")
+			return
+		}
```

Comment:
Just for review purposes, this aligns with the `gh issue create` logic here:

https://github.com/cli/cli/blob/86964d8809bed7bfbd798d444d5923b8a2bbd4d6/pkg/cmd/issue/create/create.go#L309-L333

One thing that stands out is `gh issue create` logic falls back into this situation and generates an error if title is blank regardless if it was in editor mode 🤔

Reply:
I want to try capturing my thoughts to see if I'm reasoning about this the right way:

`gh issue create` determines whether it needs to interact with the user if 1) the editor mode isn't being used and 2) either the title or body is not provided.

https://github.com/cli/cli/blob/86964d8809bed7bfbd798d444d5923b8a2bbd4d6/pkg/cmd/issue/create/create.go#L117

If it needs to interact with the user, then we have similar code to what happens on line 379 here; again, we're missing information and not using the editor:

https://github.com/cli/cli/blob/86964d8809bed7bfbd798d444d5923b8a2bbd4d6/pkg/cmd/issue/create/create.go#L232-L240

Otherwise, we're either 1) using editor mode or 2) not using the editor but have both a title and body.  Either way, both of these situations need to ensure there is a title at the least:

https://github.com/cli/cli/blob/86964d8809bed7bfbd798d444d5923b8a2bbd4d6/pkg/cmd/issue/create/create.go#L309-L333

Reply:
@andyfeller Thanks for the review🚀
When the code enters to this editor block, I can ensure that Editor mode is on.
https://github.com/cli/cli/blob/e18c00228314340509ec0c56f15a992a2950ba09/pkg/cmd/pr/create/create.go#L358-L378

Otherwise, it will enter the `else` block, which represents normal interactive mode. In this case, 1) Editor mode isn't on, and 2) either the title or body is not provided, or neither is provided. If both of title and body are provided, it will enter the block at [L336](https://github.com/cli/cli/blob/e18c00228314340509ec0c56f15a992a2950ba09/pkg/cmd/pr/create/create.go#L336) and won't go to this `else` block.

https://github.com/cli/cli/blob/e18c00228314340509ec0c56f15a992a2950ba09/pkg/cmd/pr/create/create.go#L379-L416

---
> Either way, both of these situations need to ensure there is a title at the least

Both of these two situations ensure that there is a title. Did I miss something?

Reply:
In your PR, these 4 lines checking the title are only entered if in editor mode:

https://github.com/cli/cli/blob/e18c00228314340509ec0c56f15a992a2950ba09/pkg/cmd/pr/create/create.go#L375-L378

In `gh issue create`, those 4 lines are outside of the editor mode logic, so any non-interactive and non-editor use case would still ensure there is a title: 

https://github.com/cli/cli/blob/86964d8809bed7bfbd798d444d5923b8a2bbd4d6/pkg/cmd/issue/create/create.go#L329-L332

Reply:
@andyfeller In `issue/create`, non-interactive and non-editor modes are equivalent to `titleProvided && bodyProvided == true`. In `pr/create`, when `titleProvided && bodyProvided` is true, it won't enter this `else` block, which doesn't check if the title is provided. Instead, it will enter the block at [line 336-342](https://github.com/cli/cli/blob/e18c00228314340509ec0c56f15a992a2950ba09/pkg/cmd/pr/create/create.go#L336-L342), where the presence of a title is already ensured.
https://github.com/cli/cli/blob/e18c00228314340509ec0c56f15a992a2950ba09/pkg/cmd/pr/create/create.go#L379-L416
Therefore, the existence of a title can be ensured in all scenarios: interactive mode, editor mode, and non-interactive mode+non-editor mode (`titleProvided && bodyProvided == true`).